libs/libc/stdlib/lib_strtof.c

```diff
@@ -69,28 +71,108 @@
 #  define __FLT_MAX_EXP__ (128)
 #endif
 
+#ifdef CONFIG_HAVE_LONG_DOUBLE
+#define long_double long double
+#elif CONFIG_HAVE_DOUBLE
+#define long_double double
+#else
+#define long_double float
+#endif
+
+#define shgetc(f) (*(f)++)
+#define shunget(f) ((f)--)
+#define ifexist(a,b) if((*a) !=NULL)  {**(a) = (b);}
```

Comment:
```suggestion
#define ifexist(a,b) do { if ((*a) != NULL) {**(a) = (b);} } while (0)
```

Perhaps the duplicated macro(s) can be moved to a common header?